### ISSUE TITLE
Compileandrun command is giving the following error

### ISSUE THREAD

**Issue by nikhilreddydev** [mizlan/vim-and-cp#6]
![IMG_20211212_182814_003.jpg](https://user-images.githubusercontent.com/77785504/145713286-ccaeeea6-c34f-4bc9-a32d-2e7c7445b06d.jpg)

Command passed to the terminal via termwrapper is also assuming '&&' to be a file or directory.
@mizlan @ironsoul0

**Comment by mizlan** [comment]
Which shell?

**Comment by nikhilreddydev** [comment]
Running **Ubuntu** on wsl2,
**Zsh** with no frameworks.

**Comment by mizlan** [comment]
When you would normally run CompileAndRun, first run `:echo expand('%')`. That should print out your filename. If you don't see anything at the bottom, this is the problem. You need to have the file saved on disk (although it seems to be in your image, correct me if I'm wrong).

**Comment by nikhilreddydev** [comment]
It's expanding the filename @mizlan tho.![IMG_20211213_103917_285.jpg](https://user-images.githubusercontent.com/77785504/145755808-5a324a96-085e-48f1-9780-0252a4e74692.jpg)

I think it's the problem with **"&&"**(conditional execution).
g++ is assuming it as a filename (atleast acc to error in new term window)

**Comment by nikhilreddydev** [comment]
@mizlan I am also curious to know how that **pressing any key** to close _terminal buffer_ is working. 

I see no mappings in cp.vim file for that.
Is that your custom mapping? If so, please tell me how that works.

**Comment by mizlan** [comment]
No. That's default behavior, if you see `[finished]` in the buffer name.
